eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1

```diff
@@ -164,59 +176,86 @@ function DeployStressPackage(
-        $dockerFile = Get-ChildItem $dockerFilePath
-        $dockerBuildFolder = if ($pkg.DockerBuildDir) {
-            Join-Path $pkg.Directory $pkg.DockerBuildDir
+    $dockerFilePaths = @()
+    
+    $genValFile = Join-Path $pkg.Directory "generatedValues.yaml"
+    $genVal = Get-Content $genValFile -Raw | ConvertFrom-Yaml
+    if (Test-Path $genValFile) {
+        $scenarios = $genVal.Scenarios
+        foreach ($scenario in $scenarios) {
+            if ($scenario.image) {
+                $dockerfilePath = Join-Path $pkg.Directory $scenario.image
+                $dockerFilePath = [System.IO.Path]::GetFullPath($dockerFilePath)
+                $dockerFilePaths += $dockerFilePath
+            }
+        }
+    } else {
+        if ($pkg.Dockerfile) {
+            $dockerFilePath = Join-Path $pkg.Directory $pkg.Dockerfile
         } else {
-            $dockerFile.DirectoryName
+            $dockerFilePath = "$($pkg.Directory)/Dockerfile"
         }
-        $dockerBuildFolder = [System.IO.Path]::GetFullPath($dockerBuildFolder).Trim()
-
-        Run docker build -t $imageTag -f $dockerFile $dockerBuildFolder
-        if ($LASTEXITCODE) { return }
-
-        Write-Host "`nContainer image '$imageTag' successfully built. To run commands on the container locally:" -ForegroundColor Blue
-        Write-Host "  docker run -it $imageTag" -ForegroundColor DarkBlue
-        Write-Host "  docker run -it $imageTag <shell, e.g. 'bash' 'pwsh' 'sh'>" -ForegroundColor DarkBlue
-        Write-Host "To show installed container images:" -ForegroundColor Blue
-        Write-Host "  docker image ls" -ForegroundColor DarkBlue
-        Write-Host "To show running containers:" -ForegroundColor Blue
-        Write-Host "  docker ps" -ForegroundColor DarkBlue
-
-        Run docker push $imageTag
-        if ($LASTEXITCODE) {
-            if ($login) {
-                Write-Warning "If docker push is failing due to authentication issues, try calling this script with '-Login'"
+        $dockerFilePath = [System.IO.Path]::GetFullPath($dockerFilePath)
+        $dockerFilePaths += $dockerFilePath
+    }
+    
+
+    foreach ($dockerFilePath in $dockerFilePaths) {
+        if (!(Test-Path $dockerFilePath)) {continue}
+        $dockerfileName = ($dockerfilePath -split { $_ -in '\', '/' })[-1].ToLower()
+        $imageTag = $imageTagBase + "/${dockerfileName}:${deployId}"
+        if ($pushImages) {
+            Write-Host "Building and pushing stress test docker image '$imageTag'"
+            $dockerFile = Get-ChildItem $dockerFilePath
+            $dockerBuildFolder = if ($pkg.DockerBuildDir) {
+                Join-Path $pkg.Directory $pkg.DockerBuildDir
+            } else {
+                $dockerFile.DirectoryName
+            }
+            $dockerBuildFolder = [System.IO.Path]::GetFullPath($dockerBuildFolder).Trim()
+
+            Run docker build -t $imageTag -f $dockerFile $dockerBuildFolder
+            if ($LASTEXITCODE) { return }
```

Comment:
I think since we now have multiple files to build, we should use the `Run` helper here, that way we try to surface all possible build failures instead of exiting on the first one.